OrbitCore/Path.h

```diff
@@ -10,48 +10,47 @@
-  static std::string GetExecutablePath();
-  static std::string GetBasePath();
-  static std::string GetDllPath(bool a_Is64Bit);
-  static std::string GetDllName(bool a_Is64Bit);
-  static std::string GetFileMappingFileName();
-  static std::string GetSymbolsFileName();
-  static std::string GetCachePath();
-  static std::string GetPresetPath();
-  static std::string GetPluginPath();
-  static std::string GetCapturePath();
-  static std::string GetDumpPath();
-  static std::string GetAppDataPath();
-  static std::string GetLogFilePath();
-  static std::string GetIconsPath();
-  static void Dump();
+  [[nodiscard]] static std::string GetExecutablePath();
+  [[nodiscard]] static std::string GetExecutableDir();
+  [[nodiscard]] static std::string GetBasePath();
+  [[nodiscard]] static std::string GetDllPath(bool a_Is64Bit);
+  [[nodiscard]] static std::string GetDllName(bool a_Is64Bit);
+  [[nodiscard]] static std::string GetFileMappingFileName();
+  [[nodiscard]] static std::string GetSymbolsFileName();
+  [[nodiscard]] static std::string CreateAndGetCacheDir();
+  [[nodiscard]] static std::string CreateAndGetPresetDir();
+  [[nodiscard]] static std::string CreateAndGetPluginDir();
+  [[nodiscard]] static std::string CreateAndGetCaptureDir();
+  [[nodiscard]] static std::string CreateAndGetDumpDir();
+  [[nodiscard]] static std::string CreateAndGetOrbitAppDataDir();
+  [[nodiscard]] static std::string GetLogFilePathAndCreateDir();
+  [[nodiscard]] static std::string GetIconsPath();
 
 #ifdef __linux__
-  static std::string GetHome();
+  [[nodiscard]] static std::string GetHome();
 #endif
 
-  static std::string GetFileName(const std::string& a_FullName);
-  static std::string GetFileNameNoExt(const std::string& a_FullName);
-  static std::string StripExtension(const std::string& a_FullName);
-  static std::string GetExtension(const std::string& a_FullName);
-  static std::string GetDirectory(const std::string& a_FullName);
-  static std::string GetParentDirectory(std::string a_FullName);
-  static std::string JoinPath(const std::vector<std::string>& parts);
+  [[nodiscard]] static std::string GetFileName(const std::string& a_FullName);
+  [[nodiscard]] static std::string GetFileNameNoExt(const std::string& a_FullName);
+  [[nodiscard]] static std::string StripExtension(const std::string& a_FullName);
+  [[nodiscard]] static std::string GetExtension(const std::string& a_FullName);
+  [[nodiscard]] static std::string GetDirectory(const std::string& a_FullName);
+  [[nodiscard]] static std::string GetParentDirectory(std::string a_FullName);
+  [[nodiscard]] static std::string JoinPath(const std::vector<std::string>& parts);
 
   // TODO(159868905): This is only used by OrbitModule. Remove this as soon as
   //  OrbitModule is removed.
-  static uint64_t FileSize(const std::string& a_File);
-  static bool DirExists(const std::string& a_Dir);
+  [[nodiscard]] static uint64_t FileSize(const std::string& a_File);
```

Comment:
FileSize is not used anymore - remove it.

Reply:
And there's also `std::filesystem::file_size` in case we need it, so removed this.